extension/pprofextension/README.md

```diff
@@ -48,3 +48,60 @@ extensions:
 
 The full list of settings exposed for this exporter are documented [here](./config.go)
 with detailed sample configurations [here](./testdata/config.yaml).
+
+
+### Go Profiling with pprof basics
+
+The profiler can be used to improve a program.
+The most common usage is a CPU profile, which determines where the program spends the most time while actively consuming resources.
+After generating a profile, we can interpret it in different ways.
+Go's _pprof_ offers a text, visualization, or web-based analysis.
+To collect a meaningful profile, it should run on an idle machine and if that is not possible, it is best to generate the profile several times to get consistent results.
+
+The profiler stops the program multiple times per second and collects information (such as program counters) at that point in time.
+This is called a sample, and a profile is a collection of those samples.
+
+#### Generating a profile
+The extension enables the collection of profiling data expected by _pprof_.
+To generate a profile, include the extension in your program and run it to start the server.
+If you are using the default config, it will listen on `localhost:1777`.
+To save a CPU profile on your machine, run `go tool pprof http://localhost:1777/debug/pprof/profile\?seconds\=30`.
+This will enter the interactive mode of _pprof_, where you can analyze the profile.
+
+There are different endpoints for other types of profiles.
+For instance, the memory can be analyzed using `go tool pprof http://localhost:1777/debug/pprof/heap`.
+To see all available profiles, visit `http://localhost:1777/debug/pprof/` in your browser.
+
+#### Analyzing a profile
+
+After running the above command to save the profile, _pprof_ will enter the interactive mode.
+From here, the profiles can be analyzed.
+
+Use the command `web` to open an image of the complete call graph in your browser.
```

Comment:
May want to also mention [Graphviz](https://graphviz.org/) as a prerequisite for `web`